xikolo-common/Data/Helper/UserProfileHelper.swift

```diff
@@ -49,44 +48,62 @@ open class UserProfileHelper {
     }
 
     static func logout() {
-        prefs.removePersistentDomain(forName: Bundle.main.bundleIdentifier!)
-        prefs.synchronize()
+        UserProfileHelper.clearKeychain()
         CoreDataHelper.clearCoreDataStorage()
         self.postLoginStateChange()
     }
 
     static func isLoggedIn() -> Bool {
-        return !getToken().isEmpty
+        return !self.userToken.isEmpty
     }
 
-    static func getToken() -> String {
-        return get(.token) ?? ""
+    static func postLoginStateChange() {
+        SpineHelper.updateHttpHeaders()
+        NotificationCenter.default.post(name: NotificationKeys.loginStateChangedKey, object: nil)
     }
 
-    static func save(_ key: UserDefaultsKeys.UserProfileKey, withValue value: String) {
-        prefs.set(value, forKey: key.rawValue)
-        prefs.synchronize()
-    }
+}
 
-    static func get(_ key: UserDefaultsKeys.UserProfileKey) -> String? {
-        return prefs.string(forKey: key.rawValue)
-    }
+extension UserProfileHelper {
 
-    static func getUserId() -> String? {
-        return self.get(.user)
+    private static let keychain = Keychain(service: "de.xikolo.ios").accessibility(.afterFirstUnlock)
+    private enum KeychainKey : String {
+        case userId = "de.xikolo.ios.user-id"
+        case userToken = "de.xikolo.ios.user-token"
     }
 
-    static func saveToken(_ token: String) {
-        save(.token, withValue: token)
+    static var userId: String? {
+        get {
+            return self.keychain[KeychainKey.userId.rawValue]
+        }
+        set {
+            self.keychain[KeychainKey.userId.rawValue] = newValue
+        }
     }
 
-    static func saveId(_ id: String) {
-        save(.user, withValue: id)
+    static var userToken: String {
+        get {
+            return self.keychain[KeychainKey.userToken.rawValue] ?? ""
+        }
+        set {
+            self.keychain[KeychainKey.userToken.rawValue] = newValue
+        }
     }
 
-    static func postLoginStateChange() {
-        SpineHelper.updateHttpHeaders()
-        NotificationCenter.default.post(name: NotificationKeys.loginStateChangedKey, object: nil)
+    static func clearKeychain() {
+        do {
+            try self.keychain.removeAll()
+        } catch {
+            print("Failed to clear keychain - \(error)")
+        }
     }
 
+    static func migrateLegacyKeychain() {
+        if let legacyUserId = UserDefaults.standard.string(forKey: "user"), !legacyUserId.isEmpty {
+            self.userId = legacyUserId
```

Comment:
shouldn't we delete the userDefaults after reading?

Reply:
That's right.